vendor/bat-native-ledger/src/bat/ledger/internal/bat_contribution.cc

```diff
@@ -92,6 +94,100 @@ std::string BatContribution::GetAnonizeProof(
   return proof;
 }
 
+void BatContribution::HasSufficientBalance(
+    ledger::HasSufficientBalanceToReconcileCallback callback) {
+  ledger::ActivityInfoFilter filter = ledger_->CreateActivityFilter(
+      std::string(),
+      ledger::EXCLUDE_FILTER::FILTER_ALL_EXCEPT_EXCLUDED,
+      true,
+      ledger_->GetReconcileStamp(),
+      ledger_->GetPublisherAllowNonVerified(),
+      ledger_->GetPublisherMinVisits());
+  ledger_->GetActivityInfoList(
+      0,
+      0,
+      filter,
+      std::bind(&BatContribution::GetVerifiedAutoAmount,
+                this,
+                _1,
+                _2,
+                callback));
+}
+
+void BatContribution::GetVerifiedAutoAmount(
+    const ledger::PublisherInfoList& publisher_list,
+    uint32_t record,
+    ledger::HasSufficientBalanceToReconcileCallback callback) {
+  double ac_amount = ledger_->GetContributionAmount();
+  double total_reconcile_amount(GetAmountFromVerifiedAuto(
+      publisher_list, ac_amount));
+  ledger_->GetRecurringTips(
+      std::bind(&BatContribution::GetVerifiedRecurringAmount,
+                this,
+                _1,
+                _2,
+                total_reconcile_amount,
+                publisher_list.empty(),
+                callback));
+}
+
+// static
+double BatContribution::GetAmountFromVerifiedAuto(
+    const ledger::PublisherInfoList& publisher_list,
+    double ac_amount) {
+  double non_verified_bat = 0.0;
+  for (const auto& publisher : publisher_list) {
+    if (!publisher->verified) {
+      non_verified_bat += (publisher->percent / 100.0) * ac_amount;
+    }
+  }
+  return ac_amount - non_verified_bat;
+}
+
+void BatContribution::GetVerifiedRecurringAmount(
+    const ledger::PublisherInfoList& publisher_list,
+    uint32_t record,
+    double total_reconcile_amount,
+    bool is_ac_list_empty,
+    ledger::HasSufficientBalanceToReconcileCallback callback) {
+  if (!publisher_list.empty()) {
+    total_reconcile_amount += GetAmountFromVerifiedRecurring(publisher_list);
+  }
+  if (publisher_list.empty() && is_ac_list_empty) {
+    callback(true);
+    return;
+  }
+  ledger_->FetchWalletProperties(
+      std::bind(&BatContribution::OnFetchWalletProperties,
+        this, _1, _2, total_reconcile_amount, callback));
+}
+
+void BatContribution::OnFetchWalletProperties(
```

Comment:
This one should be named more `HasSufficientBalance` specific as we will probably had more `FetchWalletProperties` calls. Maybe something like `OnSufficientBalanceWallet`?

Reply:
Done.